Core/MIPS/ARM64/Arm64RegCache.cpp

```diff
@@ -440,6 +440,9 @@ Arm64Gen::ARM64Reg Arm64RegCache::MapRegAsPointer(MIPSGPReg reg) {
 		if (!jo_->enablePointerify) {
 			// Convert to a pointer by adding the base and clearing off the top bits.
 			// If SP, we can probably avoid the top bit clear, let's play with that later.
+#ifdef MASKED_PSP_MEMORY
+			emit_->ANDI2R(EncodeRegTo64(a), EncodeRegTo64(a), 0x3FFFFFFF);
```

Comment:
This particular change isn't quite as safe, since we sub things to convert back:

https://github.com/m4xw/ppsspp/blob/b9352354c9b83c63214ce6ccdf359c05a031784e/Core/MIPS/ARM64/Arm64RegCache.cpp#L370
https://github.com/m4xw/ppsspp/blob/b9352354c9b83c63214ce6ccdf359c05a031784e/Core/MIPS/ARM64/Arm64RegCache.cpp#L404
https://github.com/m4xw/ppsspp/blob/b9352354c9b83c63214ce6ccdf359c05a031784e/Core/MIPS/ARM64/Arm64RegCache.cpp#L532
https://github.com/m4xw/ppsspp/blob/b9352354c9b83c63214ce6ccdf359c05a031784e/Core/MIPS/ARM64/Arm64RegCache.cpp#L553
https://github.com/m4xw/ppsspp/blob/b9352354c9b83c63214ce6ccdf359c05a031784e/Core/MIPS/ARM64/Arm64RegCache.cpp#L669
https://github.com/m4xw/ppsspp/blob/b9352354c9b83c63214ce6ccdf359c05a031784e/Core/MIPS/ARM64/Arm64RegCache.cpp#L761

You'll notice the arm32 code is more likely to discard values.  Also this part will need to be disabled:
https://github.com/m4xw/ppsspp/blob/b9352354c9b83c63214ce6ccdf359c05a031784e/Core/MIPS/ARM64/Arm64CompALU.cpp#L89

Otherwise, the masked value may become exposed to the game, which could cause bugs in games (i.e. if they compare pointers like `while (ptr < endptr) { *ptr++ = 0; }`.)

-[Unknown]

Reply:
Oh, hm, that's right...

Reply:
Just for clarity on resolving this, when using masked memory:

 * Need to [STR before masking here](https://github.com/m4xw/ppsspp/blob/b9352354c9b83c63214ce6ccdf359c05a031784e/Core/MIPS/ARM64/Arm64RegCache.cpp#L444), otherwise the above changes will cause major problems - `emit_->STR(INDEX_UNSIGNED, a, CTXREG, GetMipsRegOffset(reg));` or similar.
 * SUB should probably just be replaced by LDR (after adding the above STR fix.)
 * The optimization in Arm64CompALU should just be skipped (fall back to CompImmLogic case if MASKED_PSP_MEMORY.)
 * DiscardR would then also just LDR the reg, since MapRegAsPointer would STR even it.

I don't have a way to test these changes, but will probably resolve at least some crashes @m4xw.

-[Unknown]